File: benchmarks/cpp/json_like.cpp
#include <cstdio>
#include <vector>
#include <unordered_map>

struct Rec {
    long long id, score;
    std::vector<long long> tags;
};

int main() {
    const int n = 3000;
    std::vector<Rec> records;
    records.reserve(n);
    for (int i = 0; i < n; i++) {
        Rec r;
        r.id = i;
        r.score = i * 2;
        r.tags = {i % 3, i % 5, i % 7};
        records.push_back(std::move(r));
    }
    long long total = 0;
    for (auto& rec : records) {
        total += rec.id + rec.score;
        for (auto t : rec.tags) total += t;
    }
    printf("%lld\n", total);
}

File: benchmarks/json_like.py
n = 3000
records = []
i = 0
while i < n:
    rec = {"id": i, "score": i * 2, "tags": [i % 3, i % 5, i % 7]}
    records.append(rec)
    i = i + 1
total = 0
for rec in records:
    total = total + rec["id"] + rec["score"]
    for t in rec["tags"]:
        total = total + t
print(total)
